### PR TITLE
Modify Package.appxmanifest templates to include placeholder values and overwrite these values in the GeneratePackageAppxManifest task

### PR DESCRIPTION
**Pull request by japarson**
### Problem
When users attempt to open the MSIX Packaging Wizard, they are being prompted with the following error message:
image

This error message is appearing because the MAUI Package.appxmanifest templates are missing several values that the [ManifestValidator](https://devdiv.visualstudio.com/DevDiv/_git/VS?path=/src/vsproject/PackageAndDeploy/AppxManifestDesigner.UAP/ManifestData/Manager/Validation/ManifestValidator.cs) expects to be there.

We can see the missing values by opening the Package.appxmanifest editor:
image

### Solution
This PR introduces changes to the MAUI Package.appmanifest templates that ensure the file can be successfully validated. This is accomplished through the addition of "placeholder" values in the manifest templates. I've also modified the `GeneratePackageAppxManifest` task so that these values are overwritten appropriately. What this means is that if the task finds the placeholder value in the manifest, it ignores it in favor of the values from the project file.

For most of the manifest properties, we could simply use the value `$placeholder$`, which is an existing convention in the template. However, there are three scenarios where using this value is not possible:
- **Package Name:** For this property, the validator will not accept special characters such as `$`. Therefore, I chose a value that is so unique that we can reasonably expect no one will ever use it: `maui-package-name-placeholder`.
- **Package Version:** For this property, the validator will not accept anything other than a version number, e.g. `1.0.0.0`. Therefore, I chose the value `0.0.0.0`, which I believe is not likely to be a common version number. An alternative we can use could be something like `999.999.999.999`.
- **Png Files:** For the logo and splash screen properties, the validator expects the suffix `.png`. Therefore, I created a placeholder value of `$placeholder$.png`. It is also important to note that the validator does not validate that these "file paths" actually exist.

### Issues Fixed
Fixes [Bug 1526392: Unable to Publish Windows MAUI app from VS when using generated Package.appxmanifest](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1526392)

**Note:** The goal is to get this fix into 17.3 Preview 1 if possible, 17.3 Preview 1.1 if not.

### CODE REVIEW

**Review by javiercn — APPROVED** [review]

**Review by mariaghiondea — APPROVED** [review]

**Review by sbanni — APPROVED** [review]

File: src/SingleProject/Resizetizer/src/GeneratePackageAppxManifest.cs [hunk 1]
```diff
@@ -202,34 +207,26 @@ void UpdateManifest(XDocument appx)
...
 					}
 				}
 
-				// BackgroundColor=""
-				{
-					var xname = "BackgroundColor";
-					var attr = visual.Attribute(xname);
-					if (attr == null || string.IsNullOrEmpty(attr.Value))
-						visual.SetAttributeValue(xname, "transparent");
-				}
```
Comment (mattleibow) [inline src/SingleProject/Resizetizer/src/GeneratePackageAppxManifest.cs:224]: Was this for the tile background? Is this not needed for win10?

Comment (japarson) [reply src/SingleProject/Resizetizer/src/GeneratePackageAppxManifest.cs:224]: Oh, sorry. I forgot to mention in the PR that I moved that directly to the templates.

Comment (japarson) [reply src/SingleProject/Resizetizer/src/GeneratePackageAppxManifest.cs:224]: image

Comment (mattleibow) [reply src/SingleProject/Resizetizer/src/GeneratePackageAppxManifest.cs:224]: lol

Ok, so maybe have it in both places. In the templates for the UI and then the task to put it back if there is a missing value from older projects.

**Review by mattleibow — APPROVED** [review]

**Pull request merged**